src/aead/mod.rs

```diff
@@ -58,8 +59,33 @@ impl OpeningKey {
     #[inline]
     pub fn new(algorithm: &'static Algorithm, key_bytes: &[u8])
                -> Result<OpeningKey, error::Unspecified> {
+        Self::with_auth_tag_len(algorithm, key_bytes, 0)
+    }
+
+    /// Create a new opening key with authentication tag length.
+    ///
+    /// `key_bytes` must be exactly `algorithm.key_len` bytes long.
+    /// `auth_tag_len` may be truncated by passing a tag length.
+    /// A tag length of zero indicates the default tag length should be used.
+    ///
+    /// C analogs: `EVP_AEAD_CTX_init_with_direction` with direction
+    ///            `evp_aead_open`, `EVP_AEAD_CTX_init`.
+    ///
+    /// Go analog:
+    ///   [`crypto.aes.NewCipher`](https://golang.org/pkg/crypto/aes/#NewCipher)
+    /// + [`crypto.cipher.NewGCM`](https://golang.org/pkg/crypto/cipher/#NewGCM)
+    #[inline]
+    pub fn with_auth_tag_len(algorithm: &'static Algorithm, key_bytes: &[u8], auth_tag_len: usize)
```

Comment:
Instead of doing this as new constructors, create new AES_128_GCM_TRUNCATED_TAG_96 and CHACHA_POLY1305_TRUNCATED_TAG_96 algorithms.